### PR TITLE
Add vlan and mtu parameters to OVNOverlayNetworkAttachmentDefinition

### PR DESCRIPTION
**Pull request by Anatw**
Needed for the localnet feature.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Comment by Anatw** [comment]
/verified
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/

**Comment by redhat-qe-bot2** [comment]
Label job is not a predefined one, will not be added / removed.
Available labels:

 * hold
 * verified
 * wip
 * lgtm

**Comment by redhat-qe-bot2** [comment]
Label cnv-tests-runner-containerized is not a predefined one, will not be added / removed.
Available labels:

 * hold
 * verified
 * wip
 * lgtm

**Comment by redhat-qe-bot2** [comment]
Label 4031 is not a predefined one, will not be added / removed.
Available labels:

 * hold
 * verified
 * wip
 * lgtm

File: ocp_resources/network_attachment_definition.py
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:218]: not needed `This field is mandatory when not using yaml file`
you don;t have optional in the docstring.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:218]: ack

**Comment by Anatw** [comment]
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4034/

**Review by rnetser — APPROVED** [review]

**Comment by Anatw** [comment]
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4038/

**Comment by myakove** [comment]
> https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/

I get `This site can’t be reached` 
What is this link? how do I can see it?

**Review by myakove — APPROVED** [review]

**Comment by Anatw** [comment]
> > https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/
> 
> I get `This site can’t be reached` What is this link? How can I open it?

This is an internal Red Hat link, I ran a module that uses this new PR and verified it.

**Comment by myakove** [comment]
> > > https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/
> > 
> > I get `This site can’t be reached` What is this link? How can I open it?
> 
> This is an internal Red Hat link, I ran a module that uses this new PR and verified it.

I don't think it's ok to post here internal links.

**Comment by Anatw** [comment]
I removed it

**Review by myakove — APPROVED** [review]
Internal links still exist in the PR comments.

**Comment by Anatw** [comment]
> Internal links still exist in the PR comments.

I deleted all the references I have added. You need to delete your comments as well since some of the links are in them and I can't delete them from there myself.

**Review by omrirh — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Review by omrirh — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by omrirh — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by omrirh — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by rnetser** [comment]
@Anatw this pr has unresolved comments; comments must be resolved before merge

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Comment by redhat-qe-bot2** [comment]
Cherry-picked PR Add vlan and mtu parameters to OVNOverlayNetworkAttachmentDefinition into v4.15

**Pull request merged**
